### PR TITLE
Use iso_c_binding kind names.

### PR DESCRIPTION
**Pull request by jrper**
This changeset attempts to deal with the issue raised in #109 by explicitly replacing the kind parameters with those from iso_c_binding in all cases.

### CODE REVIEW

File: femtools/Diagnostic_variables.F90 [hunk 1]
```diff
@@ -31,7 +31,7 @@ module diagnostic_variables
   !!< A module to calculate and output diagnostics. This replaces the .s file.
   use fldebug 
   use global_parameters, only:FIELD_NAME_LEN,OPTION_PATH_LEN, &
-& PYTHON_FUNC_LEN, int_16, integer_size, real_size
+& PYTHON_FUNC_LEN, integer_size, real_size
```
Comment (stephankramer) [inline femtools/Diagnostic_variables.F90]: Does this one need a `use iso_c_binding` as well? I'm guessing it currently inherits it from somewhere...

**Comment by stephankramer** [comment]
Great! Thanks a lot, @jrper

**Comment by jrper** [comment]
I've added the extra statement @stephankramer asked for. Could this get a builedbot queue? Not that I'm paranoid or anything.

**Comment by stephankramer** [comment]
Buildbot queue here: http://buildbot.ese.ic.ac.uk:8080/waterfall?show=c_types_fix

After that: ship it!

**Pull request merged**